src/com/nkita/practice/collections/PersonStatisticHandler.java

```diff
@@ -298,5 +298,71 @@ public void printInfo(List<Person> personList) {
 
     }
 
+    public void checkByGenderMap() {
+        List<String> male = new ArrayList<>();
+        List<String> female = new ArrayList<>();
+
+        for (Person person : persons) {
+            if (person.getGender() == Gender.MALE) {
+                male.add(person.getName());
+            } else {
+                female.add(person.getName());
+            }
+        }
+
+        Map<Gender, List<String>> genderListMapMale = new HashMap<>();
+        Map<Gender, List<String>> genderListMapFemale = new HashMap<>();
+
+        genderListMapMale.put(Gender.MALE, male);
+        genderListMapFemale.put(Gender.FEMALE, female);
+
+        if (female.size() > 0 && male.size() > 0) {
+            System.out.println(genderListMapFemale + " " + genderListMapMale);
+        } else if (female.size() != 0) {
+            System.out.println(genderListMapFemale);
+        } else if (male.size() != 0) {
+            System.out.println(genderListMapMale);
+        } else {
+            System.out.println("Statistics is empty");
+        }
+
+    }
+
+    public void checkMapByPersonProperty(PersonProperty personProperty) {
+        Map<String, List<Person>> stringListMap = new HashMap<>();
+
+        for (Person person : persons) {
+            String key = generateKey(personProperty, person);
+
+            List<Person> personList = new ArrayList<>();
+            personList.add(person);
```

Comment:
This list will be created every time for each 'Person', but it should be created only when key is not found, so please move this logic to the 'else' block.